pkg/kn/commands/source/apiserver/update.go

```diff
@@ -74,11 +74,19 @@ func NewAPIServerUpdateCommand(p *commands.KnParams) *cobra.Command {
 			}
 
 			if cmd.Flags().Changed("resource") {
-				resources, err := apiServerUpdateFlags.GetAPIServerResourceArray()
+				added, removed, err := apiServerUpdateFlags.getUpdateAPIServerResourceArray()
 				if err != nil {
 					return err
 				}
-				b.Resources(*resources)
+				for _, k := range removed {
+					_, err = b.RemoveResource(k.ApiVersion, k.Kind, k.IsController)
+					if err != nil {
+						return err
+					}
+				}
+				for _, k := range added {
+					b.AddResource(k.ApiVersion, k.Kind, k.IsController)
+				}
```

Comment:
I would follow the same scheme here as we had for trigger filters in #603 , I think it covers the same use case.

Reply:
ok. will do.

Reply:
Resource is not a string map, but an array. I cannot leverage stringmap here.
I have to think about how to handle.

Reply:
Ah, right. Unfortunately, there are no generics in golang, that would make things much easier. Its also not easy to simulate with interfaces ;(